library/utils.py

```diff
@@ -52,3 +55,25 @@ def format_url(path: str, subpath: str) -> str:
         return url.replace("s3://", "/vsis3/")
 
     return url
+
+def get_execution_details():
+    def try_func(func):
+        try: return func()
+        except: return "could not parse"
```

Comment:
maybe this function + lambda isn't needed if all  uses of `os.environ["ENV_VAR"]` were replaced with something like `os.environ.get("ENV_VAR", "could not parse")` ?

that `dict.get("ENV_VAR")` approach let's you set a default value if the key isn't found, rather than the `KeyError` we get when using `dict["ENV_VAR"]`

Reply:
Yeah, the main reason for the alternative was the one row with 3 or 4 os.environ['thing'] and to just use the same wrapper for the function getting git username. Happy to tweak

Reply:
Left for the two noted cases above, used .get() with default for the simpler cases